Source/JavaScriptCore/runtime/RegExpConstructor.cpp

```diff
@@ -155,7 +156,8 @@ JSC_DEFINE_HOST_FUNCTION(regExpConstructorEscape, (JSGlobalObject* globalObject,
             break;
         }
 
-        if (StringView(",-=<>#&!%:;@~'`\""_s).contains(codePoint) || isStrWhiteSpace(codePoint) || U16_IS_SURROGATE(codePoint)) {
+        // FIXME: This cast to char16_t is probably not correct.
+        if (StringView(",-=<>#&!%:;@~'`\""_s).contains(static_cast<char16_t>(codePoint)) || isStrWhiteSpace(codePoint) || U16_IS_SURROGATE(codePoint)) {
```

Comment:
Not only is the cast incorrect, but I think other code on the line is, too.
```suggestion
        if (StringView(",-=<>#&!%:;@~'`\""_s).contains(static_cast<char16_t>(codePoint)) || isStrWhiteSpace(codePoint) || !U_IS_BMP(codePoint)) {
```
It’s not correct to call `U16_IS_SURROGATE` on a `char32_t`. This makes me want to wrap more macros in functions so they can type check.

Reply:
Good catch.

It would be easy to provide function alternatives for `U16_IS_SURROGATE`, `U16_LEAD`, and `U16_TRAIL`, but it's going to be hard to enforce their use. We could plausibly enforce it using a style check rule?

I won't work on this myself, but I will change it to `!U_IS_BMP` so we have one fewer thing to fix.

Reply:
When replacing `U16_GET()` with a typed version, I found that, in `Lexer<T>::lexWithoutClearingLineTerminator` in Lexer.cpp, where T is the character type, we call `U16_GET()` without verifying the type of the character. :( So yeah, types are good....